content/en/blog/_posts/2021-04-16-volume-health-monitoring-alpha.md

```diff
@@ -0,0 +1,87 @@
+
+With Volume Health Monitoring, Kubernetes could communicate with the CSI driver to retrieve any errors detected by the underlying storage system. Kubernetes can report an event about this PVC so that user can inquire this information and decide how to handle them. For example, if the volume is out of capacity, user can request a volume expansion to get more space. Volume health monitoring is a very important tool that would be beneficial to Kubernetes users.
+
+## What is Volume Health Monitoring?
+
+CSI Volume Health Monitoring allows CSI Drivers to detect abnormal volume conditions from the underlying storage systems and report them as events on PVCs or Pods.
+
+The Kubernetes components that monitor the volumes and report events with volume health information include the following:
+
+* In addition to gathering existing volume stats, Kubelet also watches volume health of the PVCs on that node. If a PVC has an abnormal health condition, an event will to reported on the pod object that is using the PVC. If multiple pods are using the same PVC, events will be reported on multiple pods.
+* An [external volume health monitor controller](https://github.com/kubernetes-csi/external-health-monitor) watches volume health of the PVCs and reports events on the PVCs.
+
+Note that the node side volume health monitoring logic was an external agent when this feature was first introduced in Kubernetes 1.19 release. In Kubernetes 1.21 release, the node side volume health monitoring logic was moved from the external agent to Kubelet to avoid duplicate CSI function calls. With this change in 1.21, a new alpha feature gate `CSIVolumeHealth` was introduced for the volume health monitoring logic in Kubelet.
+
+Currently the Volume Health Monitoring feature is informational only as it  only reports abnormal volume health events on PVCs or Pods. Users will need to check these events and manually fix the problems. This feature serves as a stepping stone towards programmatic detection and resolution of volume health issues by Kubernetes in the future.
+
+## How do I use Volume Health on my Kubernetes Cluster?
+
+To learn how to use External Volume Health Monitoring Controller, see details in this [doc](
```

Comment:
This sentence should be under the storage vendor part

Reply:
Ok